Src/FluentAssertions/Xml/XmlReaderValidator.cs

```diff
@@ -103,8 +104,41 @@ private ValidationResult Validate()
                     return validationResult;
                 }
 
-                subjectReader.Read();
-                otherReader.Read();
+                var readOnSubjectReader = true;
+                var readOnOtherReader = true;
+                if (subjectReader.IsEmptyElement && !otherReader.IsEmptyElement && !otherReader.HasValue)
```

Comment:
You could make this code easier to read if we wrap the `XmlReader` in another class that exposes specific methods and properties instead of directly interacting with the `XmlReader`

Reply:
I understand what you mean. But, I think it is not really possible.
Both readers must get synchronized. (Advancing "subject reader" is only done in specific state of "other reader" and vice versa.

Reply:
I get that, but wrapping the reader you can make that same code much more intention-revealing.

Reply:
Please have a look on the new version.